### PR TITLE
Issue#1189 : added documentation of CheckAndMutate to modify a value only if it exists

### PR DESCRIPTION
**Pull request by roopak-qlogic**
Fixes issue #1189

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/1201)

### CODE REVIEW

**Comment by roopak-qlogic** [comment]
fixing build issue

**Comment by roopak-qlogic** [comment]
Yes, after raising PR, even I saw similar changes has been already done in PR #1199, but I can see small differences in file /bigtable-samples-data-client.dox.
So, now we can do  wait for approving PR #1199, and then update my small changes.

File: google/cloud/bigtable/doc/bigtable-samples-data-client.dox [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+@snippet data_admin_snippets.cc bigtable includes
+
+### Connect to the Cloud Bigtable data client endpoint.
+
+@snippet data_admin_snippets.cc connect data 
+
+### Use CheckAndMutate to modify a value only if it exists.
+
+@snippet data_admin_snippets.cc check and mutate
```
Comment (sduskis) [inline google/cloud/bigtable/doc/bigtable-samples-data-client.dox]: Check and Mutate is in data, not admin.

**Comment by coryan** [comment]
@roopak-qlogic please rebase so we can see the actual changes in this PR.

**Comment by roopak-qlogic** [comment]
rebase is done, and  also fixed review comment.

**Comment by coryan** [comment]
@sduskis This LGTM, are you Okay if we merge it?

**Pull request merged**